### ISSUE TITLE
Out of Memory Exception: `TabControl.WmReflectDrawItem()`

### ISSUE THREAD

**Issue by samtrion** [dotnet/winforms#12157]
### .NET version

.NET 8.0
SDK: 8.0.204/8.0.401
RUNTIME: 8.0.8

### Did it work in .NET Framework?

Not tested/verified

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, it's related to #10534

### Issue description

In our application, we receive a `OutOfMemoryException` on a daily base. Roughly 4000 times in the last 90 days.

```txt
Error: Out of memory.

   at System.Windows.Forms.TabControl.WmReflectDrawItem(Message& m)
   at System.Windows.Forms.TabControl.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam)
```

Please merge #10724, which addresses the same problem as before for `ComboBox` and `ListBox`.

### Steps to reproduce

---

**Comment by merriemcgaw** [comment]
@JeremyKuhne we should take this (or a similar fix)

**Comment by merriemcgaw** [comment]
@lonitra can you take this to Tactics next week?

**Comment by samtrion** [comment]
@lonitra Is there already a planned delivery date for this?

**Comment by guyvaio** [comment]
+1

**Comment by lonitra** [comment]
> Is there already a planned delivery date for this?

This will be shipped as part of 8.0.11 release, which I believe will be November 5
